k8s/src/main/scala/io/buoyant/k8s/IngressCache.scala

```diff
@@ -31,34 +33,67 @@ case class IngressSpec(
 case class IngressPath(
   host: Option[String] = None,
   path: Option[String] = None,
+  useCapturingGroups: Option[Boolean],
```

Comment:
it means that the annotation is not present and it's necessary to support this requirement:
>If useCapturingGroupsGlobal is true but useCapturingGroups is false, the whole thing evaluates to true which seems potentially confusing. I'd either default to useCapturingGroups if defined (since it's a more local scope), or remove the capture group settings from the config altogether.

I think the code is self explanatory also haven't seen any comments in way more supprising parts of code